.test-conda-env-py3.yml

```diff
@@ -13,4 +13,4 @@ dependencies:
 - pyopencl
 - islpy
 - pip
-- jax
```

Comment:
Do we still need this? I suspect we want to be on the "new" side of their [`Array` migration](https://jax.readthedocs.io/en/latest/jax_array_migration.html), and have version constraints to match.

Reply:
I kept the version restriction on jax in this PR for the moment to see if jax 0.3 can handle the small code changes in this  PR as well. If so, we might not need a version constraint at all.